src/target/source/codegen_c.cc

```diff
@@ -648,6 +648,21 @@ void CodeGenC::VisitExpr_(const CallNode* op, std::ostream& os) {  // NOLINT(*)
       os << " != ";
       this->PrintExpr(op->args[0], os);
       os << ")";
+    } else if (op->op.as<OpNode>()->name == "tir.round") {
+      CHECK_EQ(op->args.size(), 1U);
```

Comment:
We should ideally handle these intriniscs lowering stage before the codegen. The main reason is we have not yet registered the dispatch rule for C backend, or default

- LLVM intrinsic dispatch rule: https://github.com/apache/tvm/blob/main/src/target/llvm/intrin_rule_llvm.cc#L84
- https://github.com/apache/tvm/blob/main/src/target/intrin_rule.cc#L54

Reply:
cc other reviewers so they are also aware of the case, cc @areusch @manupa-arm

Reply:
@tqchen are you suggesting we should just redirect this to DispatchPureExtern via intrinsic lowering ? If so yes, sounds like a good idea.

Reply:
* Addedd missing to ```target/intrin_rule.cc```.
* Added test cases to check C lowering.
* In C generated code we get proper C function calls:
```
((float*)B)[(i0)] = ceilf(((float*)A)[(i0)]);
((float*)B)[(i0)] = floorf(((float*)A)[(i0)]);
((float*)B)[(i0)] = roundf(((float*)A)[(i0)]);
```